apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -251,6 +266,86 @@ func (m *ModelManagerAPI) checkAddModelPermission(cloud string, userTag names.Us
 	return true, nil
 }
 
+// createModelNew is the work in progress logic for moving this facade over to
+// the new services layer. It should be considered the new logic that will be
+// merged in place of state eventually. We have split it out as a temp work
+// around to get the DDL changes needed into Juju before finishing the rest.
+func (m *ModelManagerAPI) createModelNew(ctx context.Context, uuid string, args params.ModelCreateArgs) error {
```

Comment:
So most of what's below will all eventually move to the services layer?

Reply:
No, this is all staying. It can't move to the services layer because it is tangled in permission logic.